arch: add boot parameters in x86 arch and  initrd definition and function

studychao opened the pull request:
add boot parameters to help define x86 guest boot information

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

add initrd definition and function that could help to find load address
for the initrd

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
arch+boot: add boot parameters in x86 arch

add boot parameters to help define x86 guest boot information

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

jiangliu commented on crates/db-boot/src/x86_64/mod.rs:
```diff
@@ -3,8 +3,9 @@
 
 //! VM boot related constants and utilities for `x86_64` architecture.
 
+use super::PAGE_SIZE;
```
move it near to line 10? They both imports from the crate itself.

studychao replied:
changed.

jiangliu commented on crates/db-boot/src/lib.rs:
```diff
@@ -17,3 +17,14 @@ pub use x86_64::*;
+}
+
+/// Default (smallest) memory page size for the supported architectures.
+pub const PAGE_SIZE: usize = 4096;
```
PAGE_SIZE should be defined per arch instead of a generic value. Some arch even have variable page size.

jiangliu replied:
Move it into x86_64/mod.rs?

studychao replied:
moved

jiangliu commented on crates/db-boot/src/x86_64/mod.rs:
```diff
@@ -17,6 +17,21 @@ pub mod mpspec;
+// is prohibited.
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub(crate) struct BootParamsWrapper(bootparam::boot_params);
```
It's `pub(crate)` but not used, is it by design and will be used later?

studychao replied:
`(crate)` removed

jiangliu commented on crates/db-boot/src/x86_64/bootparam.rs:
```diff
@@ -0,0 +1,3876 @@
+// found in the THIRD-PARTY file.
+
+/*
+ * automatically generated by rust-bindgen
```
There are clippy warnings, so seems we need to regenerate the file with latest bindgen.

studychao replied:
sure, let me regenerate it.

studychao replied:
Done

studychao replied:
seems like latest version of bindgen still has this kind of clippy problem in the `test` area. I'll manually change it.

studychao replied:
changed.

studychao committed:
boot: add initrd support

add initrd definition and function that could help to find load address
for the initrd.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

jiangliu commented on crates/db-boot/src/x86_64/bootparam.rs:
```diff
@@ -0,0 +1,4619 @@
+// Copyright 2021-2022 Alibaba Cloud. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
```
Line 2 is redundant

studychao committed:
boot: remove redundant license line

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
boot: add boot param related UT

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
CI: ignore rust-bindgen file

Since rust-bindgen doesn't have good ut coverage, we ignore the file
generated by rust-bindgen in codecov

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>
